src/pymovements/gaze/io.py

```diff
@@ -268,21 +268,25 @@ def from_csv(
 def from_asc(
         file: str | Path,
         *,
-        patterns: str | list | None = 'eyelink',
+        patterns: str | list[dict[str, Any] | str] | None = 'eyelink',
+        metadata_patterns: list[dict[str, Any] | str] | None = None,
         schema: dict[str, Any] | None = None,
         experiment: Experiment | None = None,
         add_columns: dict[str, str] | None = None,
         column_dtypes: dict[str, Any] | None = None,
-) -> GazeDataFrame:
+) -> tuple[GazeDataFrame, dict[str, Any]]:
     """Initialize a :py:class:`pymovements.gaze.gaze_dataframe.GazeDataFrame`.
 
     Parameters
     ----------
     file: str | Path
         Path of IPC/feather file.
-    patterns: str | list | None
+    patterns: str | list[dict[str, Any] | str] | None
         list of patterns to match for additional columns or a key identifier of eye tracker specific
```

Comment:
List should be capitalized in the beginning. Additionally we might need and example for a list of dict?